code/modules/reagents/oldchem/reagents/_reagent_base.dm

```diff
@@ -69,6 +69,9 @@
 		holder.remove_reagent(id, metabolization_rate) //By default it slowly disappears.
 		current_cycle++
 
+/datum/reagent/proc/on_mob_death(var/mob/living/M as mob, var/alien)	//use this to have chems have a "death-triggered" effect
```

Comment:
still need to get rid of teh var here and as mob.

Also the alien argument is completely dead now.